ex2_hal/adcs/equipment_handler/src/adcs_io.c

```diff
@@ -225,6 +229,10 @@ ADCS_returnState request_uart_telemetry(uint8_t TM_ID, uint8_t *telemetry, uint3
 
     // Destuff the reply
     uint8_t *thin_reply = (uint8_t *)pvPortMalloc((received - ADCS_TM_HEADER_SZ) * sizeof(uint8_t));
+    if (thin_reply == NULL){
+        vPortFree(reply);
+        return ADCS_MALLOC_FAILED;
```

Comment:
missing semaphore give. Actually, check all the paths for semphore handling

Reply:
Yeah there were two more